### PR TITLE
Allow to load secrets from azure credential file

### PR DESCRIPTION
**Pull request by mkaszub**
This feature introduce loading of secrets from credential file that may be crated by calling azure cli.

Azure credential file may be crated by calling:
```bash
az ad sp create-for-rbac --sdk-auth > credentials.json
```
chaostollkit extension will pick up credential file when:
- no secrets are defined in experiment file
- path to the credential file is stored in environment variable AZURE_AUTH_LOCATION

More info about azure credential file may be found:
https://docs.microsoft.com/en-us/azure/developer/python/azure-sdk-authenticate

This should solve as well #58

### CODE REVIEW

**Comment by mkaszub** [comment]
@bugra-derre  could you take a look?

File: chaosazure/__init__.py [hunk 1]
```diff
@@ -114,29 +35,30 @@ def discover(discover_system: bool = True) -> Discovery:
...
-        configuration: Configuration) -> ComputeManagementClient:
-    with auth(secrets) as authentication:
-        _subscription_id = configuration.get("azure_subscription_id")
-        if not _subscription_id:
-            _subscription_id = configuration['azure']['subscription_id']
-        _base_url = authentication.cloud_environment.endpoints.resource_manager
+        experiment_secrets: Secrets,
+        experiment_configuration: Configuration) -> ComputeManagementClient:
 
+    secrets = load_secrets(experiment_secrets)
```
Comment (bugra-derre) [inline chaosazure/__init__.py:41]: Nicely separated. Chapeau.

File: README.md [hunk 2]
```diff
@@ -30,125 +30,172 @@ experiment file:
...
+- Azure credential file
+
+The above configuration values have the same precedence as listed.
 
 [creds]: https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-connect-to-secure-cluster
 [requests]: http://docs.python-requests.org/en/master/
 [sdk]: https://github.com/Azure/azure-sdk-for-python
 
-There are two ways of doing this:
+### Credentials
```
[the remark this answers is not in the document]
Comment (mkaszub) [reply README.md:68]: Do you mean reading secrets from experiment file: **env variable** vs **hardcoded**

I can change order of it.

**This is way I pushe it down:**
I think this is not related to this extension. This is related to the Chaos OpenAPI specification so we do not have to write it here.
https://docs.chaostoolkit.org/reference/api/experiment/#environment-secrets

In my opinion we can leave a note that it is possible (and leave a link)

[the remark this answers is not in the document]
Comment (bugra-derre) [reply README.md:68]: I did not mean the hardcoded secrets vs environment creds in particular.

You are right: You could remove the already mentioned secrets topics from here - as you already mentioned: they are already here -> https://docs.chaostoolkit.org/reference/api/experiment/#environment-secrets and that's absolutely redundant.

And yes. Focus on the specialties of this extension - together with a link to the basic and generic stuff.

[the remark this answers is not in the document]
Comment (bugra-derre) [reply README.md:68]: I like the idea you have. Remove the basic / generic secrets section - it is already written down in the chaostoolkit. There is no need to duplicate the text.

And instead focus on the specialties of this extension,

**Comment by bugra-derre** [comment]
Please add your changes in a short sentence into the **CHANGELOG.md**

**Pull request merged**